sklearn/ensemble/gradient_boosting.py

```diff
@@ -1714,8 +1714,11 @@ def feature_importances_(self):
 
         total_sum = np.zeros((self.n_features_, ), dtype=np.float64)
         for stage in self.estimators_:
+            trees = [tree for tree in stage if tree.tree_.node_count > 1]
+            if len(trees) == 0:
+                continue
             stage_sum = sum(tree.tree_.compute_feature_importances(
-                normalize=False) for tree in stage) / len(stage)
+                normalize=False) for tree in trees) / len(trees)
             total_sum += stage_sum
 
         importances = total_sum / total_sum.sum()
```

Comment:
This will give a ZeroDivision error if all the trees in the ensemble have only one node. You can reproduce with something like:

```py
X, y = make_classification(n_samples=10, random_state=1)
gbc = GradientBoostingClassifier(min_samples_leaf=10)
```

Reply:
that's why I have the `if len(trees) == 0: continue` above it.

Reply:
Yes, and if you hit `continue` for all trees then `total_sum.sum()` is 0 and you get an error